apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -286,6 +296,19 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
     setQueuedUserAction(action);
   }, []);
 
+  const generateContinueAction = useCallback(() => {
+    const sectionsToGenerate = nextSectionsToGenerate(lessonPlan);
+    if (sectionsToGenerate.length === 0) {
+      return "Continue - check the lesson plan for consistency and correctness";
+    } else {
+      const sentenceCaseSections = sectionsToGenerate
+        .map((section) => camelCaseToTitleCase(section))
+        .join(", ")
+        .replace(/, ([^,]*)$/, " and $1");
+      return `Continue. Generate the ${sentenceCaseSections} section${sectionsToGenerate.length > 1 ? "s" : ""}.`;
```

Comment:
Doing this means that the front end should be more likely to rescue from missing sections. It will automatically trigger generation of the correct next section group OR try to fill in if something is missing.